dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala

```diff
@@ -1507,8 +1511,15 @@ abstract class DLCWallet extends Wallet with AnyDLCHDWalletApi {
 
   override def executeDLCRefund(contractId: ByteVector): Future[Transaction] = {
     for {
-      (executor, setup) <- executorAndSetupFromDb(contractId)
-      refundTx = executor.executeRefundDLC(setup).refundTx
+      dlcDbOpt <- dlcDAO.findByContractId(contractId)
+      dlcDb = dlcDbOpt.get
+
+      executor <- executorFromDb(dlcDb.paramHash)
+      refundSigDbOpt <- dlcRefundSigDAO.findByParamHash(dlcDb.paramHash,
+                                                        !dlcDb.isInitiator)
+      refundSig = refundSigDbOpt.get.refundSig
+
+      refundTx = executor.executeRefundDLC(refundSig).refundTx
```

Comment:
Ideally you would be able to call
```
def completeRefundTx(
      localSig: PartialSignature,
      remoteSig: PartialSignature,
      fundingMultiSig: MultiSignatureScriptPubKey,
      fundingTx: Transaction,
      uRefundTx: WitnessTransaction): WitnessTransaction
```
from `DLCTxSigner` (the object) directly, is that possible or do we not store the txs anywhere?

Reply:
We have the funding tx but not the refund tx.

Reply:
Although I think that would be slower

Reply:
It definitely wouldn't be slower because strictly more work is currently being triggered by the call to `executorFromDb` followed by this, but I'll drop this for now because I suppose that this is currently a simpler diff and it is more generally a nice thing in this file that it should be refactored to use the case classes less in situations like these, not just in this one case